### ISSUE TITLE
UMD with 'process.env.NODE_ENV' occurence (dev/prod)

### ISSUE THREAD

**Issue by UbikZ** [i18next/react-i18next#681]
Hi there.

**Describe the bug**
Just want to use UMD build of react-i18next, but crashing because of `process.env.NODE_ENV` occurence inside development/production files.

**Occurs in react-i18next version**
https://unpkg.com/react-i18next@9.0.4/dist/umd/react-i18next.js
https://unpkg.com/react-i18next@9.0.4/dist/umd/react-i18next.min.js

**To Reproduce**
1. Go https://codepen.io/anon/pen/ZVdRNX
2. Check console

**To Fix**
1. `npm i rollup-plugin-replace -D`
2. In `rollup.config.js` : 
* Import `rollup-plugin-replace`
* Add the plugin :
```
replace({
      'process.env.NODE_ENV': JSON.stringify(compress ? 'production' : 'development'),
}),
```

Thank's.

Gabriel.

**Comment by jamuhl** [comment]
that comes from react-is which is used by hoist-non-react-statics -> https://unpkg.com/react-is@16.7.0/index.js

we guard our calls like: https://github.com/i18next/react-i18next/blob/master/src/utils.js#L16

the only thing you can do is adding:

`<script>process = { env: { NODE_ENV: 'production' } }</script>`

**Comment by UbikZ** [comment]
Sorry, but I don't get it.

* https://unpkg.com/react-is@16.7.0/index.js -> *CJS build*
* https://unpkg.com/react-is@16.7.0/umd/react-is.development.js -> *UMD build*

The `process.env.NODE_ENV` is always computed in UMD builds in order to remove some code from the build if production mode is enabled or not.

If **react-is** does not need the block code inside the `if (process.env.NODE_ENV !== "production") {` in production, your **react-i18next** production build will not needed it neither.

Am I missing something ?

**Comment by jamuhl** [comment]
Not working like that...building a module you pull in dependencies...those are commonjs independent of rollup outputting umd in the end -> therefore it ends up like it is.

If you like to create a build (hardcoded production and development) like react-is does it - feel free to provide that as a PR...would be highly appreciated

**Comment by UbikZ** [comment]
The pull-request #683 works like a charm for UMD build.
Can you check all is good for you ?
Thank's.

**Comment by jamuhl** [comment]
published in react-i18next@9.0.5
